AuctionSite.DataAccess/Repositories/Interfaces/IUserRepository.cs

```diff
@@ -1,6 +1,13 @@
-﻿namespace AuctionSite.DataAccess.Repositories.Interfaces
+﻿using System.Threading.Tasks;
+using AuctionSite.Shared.Dto;
+
+namespace AuctionSite.DataAccess.Repositories.Interfaces
 {
     public interface IUserRepository
     {
+        Task<UserDto> GetByEmailOrDefaultAsync(string email);
+        bool WithThisEmailExists(string email);
+        bool WithThisUsernameExists(string username);
```

Comment:
Linia 9 i 10: w ten sposó robi się bałagan w repository, wystarczy IQueryable<UserDto> GetAll() a to co chcemy sprawdzić można sprawdzić przy użyciu .Any(func), nie chcemy żeby repo skończyło z takim bałaganem jak w DCMT :)

Reply:
Również zależy od konwencji. Możemy zrobić tak jak mówisz